Album view adjust

Walikdik opened the pull request.

ramack commented:
thanks for your PR. As this is quite UI based (surprise surprise :-) it is hard to review it just by watching the code. Also the many commits don't make it easier. I'll have to try that one first but please give me a few days. @voriondev if you have time and want to give it a try don't hesitate to review it.

You might want to amend the commits or cancel the PR and create a fresh, clean one to simplify it if you have time. Also it seems you have added a dependency to a library. Are you sure that this is 1. needed and 2. a maintained one? And please add it in the about view (only in strings.xml) as used library.

ramack commented:
ah and don't forget to put you there as a contributor

voriondev commented on .idea/misc.xml:
```diff
@@ -39,5 +39,5 @@
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="JDK" project-jdk-type="JavaSDK" />
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" project-jdk-name="1.8" project-jdk-type="JavaSDK" />
```
This is not "needed" for us as it is related to your IDE.

voriondev commented on app/build.gradle:
```diff
@@ -124,6 +124,8 @@ dependencies {
     implementation 'com.google.guava:guava:28.1-jre'
     annotationProcessor 'com.google.guava:guava:24.1-jre'
     implementation 'org.apache.commons:commons-lang3:3.9'
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
```
Here you may want to prefer to use androidx instead of android support which is deprecated.

Here is the right one: `implementation 'androidx.constraintlayout:constraintlayout:1.1.3'`

voriondev commented on app/build.gradle:
```diff
@@ -124,6 +124,8 @@ dependencies {
     annotationProcessor 'com.google.guava:guava:24.1-jre'
     implementation 'org.apache.commons:commons-lang3:3.9'
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    implementation 'com.makeramen:roundedimageview:2.3.0'
```
Also this library is not kept up to date and there is a bunch of issues on it..

Is there no alternative ?

voriondev commented on unnamed.patch:
```diff
@@ -0,0 +1,151 @@
+Index: .idea/misc.xml
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
```
What is the purpose of this file ?

voriondev commented on app/src/main/res/layout/item_album.xml:
```diff
@@ -1,5 +1,6 @@
 <layout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto">
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:card_view="http://schemas.android.com/tools">
```
This seems to be used nowhere..

Walikdik commented:
Hi @voriondev,
 I added a new commit where : conflicts were resolved, followed your instructions and made changes, also I reworked view. Review and let me know if it is okey. Thanks

voriondev commented on app/src/main/res/layout/item_album.xml:
```diff
@@ -62,33 +77,40 @@
                 android:layout_marginTop="15dp"
                 android:textAppearance="?android:attr/textAppearanceLarge"
                 android:textColor="@color/piwigo_orange"
+                tools:text="peterdqdwqdqwdqdwqdqw"
```
This placeholder is kinda weird.. Maybe go for something more simple like "textview" ?

ramack replied:
Or even better would be to put the 'content' here... "Album title"

ramack replied:
Wow this looks nice!

> Also, we can only open an album by clicking on the picture, when we should be able to open it by clicking anywhere on the ticket.
> 
> Finally, I don't know @ramack's opinion about this, but you may want to close this PR and open another one with only the needed commits :)

Yes, I think a fray PR with just the interesting commits would be best, but for sure we can squash the PR and then it'll become a single commit during 'merge'...

voriondev commented on app/src/main/res/layout/item_album.xml:
```diff
@@ -62,33 +77,40 @@
                 app:layout_constraintBottom_toBottomOf="parent"
-                app:layout_constraintStart_toStartOf="@+id/guideline"
                 app:layout_constraintEnd_toEndOf="parent"
+                tools:text="textetxtextefwefwefwefwefwefwefwefwefwetx"
```
Same thing here, "textview2" would do the trick ;)

ramack replied:
Again, best would be to choose the description of the content here.

Walikdik commented:
> Once again thank you for your contribution.
> There are, to me, still a few bugs needing your attention.
> First of all, the image scale is pretty strange on portrait images (see screenshot below).
> 
> Also, we can only open an album by clicking on the picture, when we should be able to open it by clicking anywhere on the ticket.
> Finally, I don't know @ramack's opinion about this, but you may want to close this PR and open another one with only the needed commits :)

I can not figure out how to solve image scaling. You want square preview but this should be also issue for portrait images, am I right?

ramack commented on app/build.gradle:
```diff
@@ -124,8 +124,10 @@ dependencies {
     annotationProcessor 'com.google.guava:guava:24.1-jre'
     implementation 'org.apache.commons:commons-lang3:3.9'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'com.vipulasri:ticketview:1.0.7'
```
this lib looks at least not unmaintained... Usually I am not so happy with adding new dependencies, but why not. Doing it from scratch is also not worth the effort. But please make sure to mention it in the list of used libraries

Walikdik replied:
Constrain Layout is the most common layout now. The another one, "ticketView" is because of this semicircle.

ramack commented on app/src/main/res/layout/item_album.xml:
```diff
@@ -8,49 +9,108 @@
+                android:src="@{viewModel.url}"
+                android:contentDescription="@{viewModel.title}"
+                android:onClick="@{viewModel::onClickDo}"
+                android:scaleType="fitXY"
```
this one looks suspicious for the aspect ratio... Do you want to try it with CENTER_CROP?

Walikdik replied:
Here is look with CENTER_CROP
![piwigo3](https://user-images.githubusercontent.com/56266180/69579271-793fe200-0fd2-11ea-9d7a-de47f8c97929.PNG)

ramack commented:
this is superseded by #210 so let's discard this one here.
